nextflow_schema.json

```diff
@@ -483,6 +489,116 @@
+                    "description": "Global minimum query length of an AminoAcid strand.",
+                    "minimum": 1
+                },
+                "lx_max_dna_len": {
+                    "type": "integer",
+                    "default": 10000000,
+                    "description": "Global maximum query length of DNA strand.",
+                    "minimum": 10
+                },
+                "lx_max_aa_len": {
+                    "type": "integer",
+                    "default": 10000000,
+                    "description": "Gloval maximum query length of AminoAcid strand.",
+                    "minimum": 10
+                },
+                "lx_min_dna_ident": {
+                    "type": "number",
+                    "default": 80,
+                    "description": "Global minimum DNA percent identity required for match. (Float value)",
+                    "minimum": 0.001,
+                    "maximum": 100
+                },
+                "lx_min_aa_ident": {
+                    "type": "number",
+                    "default": 80,
+                    "description": "Global minimum minimum AminoAcid percent identiy required for match. (Float value)",
+                    "minimum": 0.001,
+                    "maximum": 100
+                },
+                "lx_min_dna_match_cov": {
+                    "type": "number",
+                    "default": 80,
+                    "description": "Gloval minimum DNA percent hit coverage identity required for match. (float)",
+                    "minimum": 0.001,
+                    "maximum": 100
+                },
+                "lx_min_aa_match_cov": {
+                    "type": "number",
+                    "default": 80,
+                    "description": "Global minimum AminoAcid hit coverage identity required for match. (float)",
+                    "minimum": 0.001,
+                    "maximum": 100
+                },
+                "lx_max_target_seqs": {
+                    "type": "number",
+                    "default": 10,
+                    "description": "Maximum number of sequence hits per a query",
+                    "minimum": 1
+                },
+                "lx_extraction_mode": {
+                    "type": "string",
+                    "default": "raw",
+                    "description": "Different ways to run locidex.",
+                    "enum": [
+                        "snps",
+                        "trim",
+                        "raw",
+                        "extend"
+                    ]
+                },
+                "lx_report_mode": {
+                    "type": "string",
+                    "default": "normal",
+                    "description": "Allele profile assignment.",
+                    "enum": [
+                        "normal",
+                        "conservative"
+                    ]
+                },
+                "lx_report_prop": {
```

Comment:
How is this used? As in, where is the metadata coming from that this label would apply to?

Since this value is written to a bash script prior to execution, leaving it as an unrestricted string leaves open the potential for someone to execute arbitrary code in the locidex report step. For example, setting `--lx_report_prop label;echo hello;`.

Could this be restricted using `pattern` and a regex to maybe alphanumeric and `_` and `-` or some other limited combination of characters?

Reply:
Added a regex expression to match only strings containing underscores, and updated the description to state what is allowed in: c8e0c8a003520471478e446d9fef5107084608ea